browser/ui/tabs/split_view_browser_data.h

```diff
@@ -39,6 +42,26 @@ class SplitViewBrowserData : public BrowserUserData<SplitViewBrowserData> {
   void AddObserver(SplitViewBrowserDataObserver* observer);
   void RemoveObserver(SplitViewBrowserDataObserver* observer);
 
+  class OnTabDragEndedClosure {
+   public:
+    OnTabDragEndedClosure();
+    OnTabDragEndedClosure(base::WeakPtr<SplitViewBrowserData> data,
+                          base::OnceClosure closure);
+    OnTabDragEndedClosure(OnTabDragEndedClosure&& other) noexcept;
+    OnTabDragEndedClosure& operator=(OnTabDragEndedClosure&& other) noexcept;
+    ~OnTabDragEndedClosure();
+
+    void RunAndReset();
+
+   private:
+    void RunCurrentClosureIfNeededAndReplaceWith(OnTabDragEndedClosure&& other);
+
+    base::WeakPtr<SplitViewBrowserData> data_;
+
+    std::unique_ptr<base::ScopedClosureRunner> closure_;
```

Comment:
need to use `unique_ptr` for scoped runner?
scoped runner is assignable and can boolean test so maybe `base::ScopedClosureRunner closure_` would be sufficient.